compiler/rustc_error_messages/locales/en-US/hir_analysis.ftl

```diff
@@ -133,3 +133,8 @@ hir_analysis_extern_crate_not_idiomatic =
     .suggestion = convert it to a `{$msg_code}`
 
 hir_analysis_expected_used_symbol = expected `used`, `used(compiler)` or `used(linker)`
+hir_analysis_expected_used_symbol = expected `used`, `used(compiler)` or `used(linker)`
+
+hir_analysis_missing_parentheses_in_range = `{$ty_str}` is not an iterator
```

Comment:
this feels wrong as your error also triggers for `0..7.contains(&3)`. Thinking about this: Instead of emitting a completely separate error, can you only add a note to the error created further down, ignoring all other attempts at adding help or notes?

Reply:
This is previous error message:
![image](https://user-images.githubusercontent.com/230646/193831599-6ca6619a-6d4b-46aa-a99b-1443ba914b07.png)

I changed it to like this:
![image](https://user-images.githubusercontent.com/230646/193831717-594dda16-2621-49f2-887f-6f445b5bbe6b.png)

Do you mean keep the `error[E0689]: can't call method `contains` on ambiguous numeric type `{integer}`` as before?

Reply:
yes, `contains` is not a method on iterators

Reply:
I change it to:
> error[E0689]: can't call method `contains` on type `{integer}`
And then add the help message.

I won't want to mix the code with `suggest_constraining_numerical_ty`, which will make the code more complicated.